### PR TITLE
Only configure requested lammps features

### LINKED ISSUE #3524: Failed to build lammps
Lammps build failed with an error message "No spec with name lapack in lammps...".   I tried to specify `openblas` dependecy to it, but spec said "".

```
$ spack purge; spack clean lammps; spack install lammps %gcc@6 ^openmpi
==> Installing lammps
==> openmpi is externally installed in /lustre/spack/tools/linux-centos7-x86_64/gcc-6.
3.0/openmpi-2.0.2-wo3vh6hf2b64jhmlbvnapkaxlhkaxc2l
==> fftw is already installed in /lustre/spack/sandybridge/linux-centos7-x86_64/gcc-6.
3.0/fftw-3.3.6-pl1-bh5rh3gb7ab4swojghpvs6gn6pq45n2j
==> Using cached archive: /tmp/rpm/spack_source_cache/lammps/lammps-2016.11.17.tar.gz
==> Staging archive: /home/rpm/spack/var/spack/stage/lammps-2016.11.17-6s45pe4moej4nci
makgqx5qzh5juxzjj/stable_17Nov2016.tar.gz
==> Created stage in /home/rpm/spack/var/spack/stage/lammps-2016.11.17-6s45pe4moej4nci
makgqx5qzh5juxzjj
==> No patches needed for lammps
==> Building lammps [MakefilePackage]
==> Executing phase : 'edit'
==> Executing phase : 'build'
==> Error: KeyError: 'No spec with name lapack in lammps@2016.11.17%gcc@6.3.0+lib+manybody~meam~rigid~user-atc~user-nc-dump~voronoi arch=linux-centos7-x86_64 ^fftw@3.3.6-pl1%gcc@6.3.0+float+long_double+mpi+openmp~pfft_patches~quad arch=linux-centos7-x86_64 ^openmpi@2.0.2%gcc@6.3.0~java~mxm+pmi~psm~psm2+slurm~sqlite3~thread_multiple~tm+verbs+vt arch=linux-centos7-x86_64'
/home/rpm/spack/var/spack/repos/builtin/packages/lammps/package.py:169, in build_user_atc:
     145      def build_user_atc(self):
     146          with working_dir('lib/atc'):
     147              filter_file(r'CC =.*',
     148                          'CC = {0}'.format(self.compiler.cxx),
     149                          'Makefile.icc')
     150
     151              mpi_include = self.spec['mpi'].prefix.include
     152
     153              filter_file(r'CCFLAGS = *',
     154                          'CCFLAGS = -I{0} '.format(mpi_include),
     155                          'Makefile.icc')
     156
     157              filter_file('LINK =.*',
     158                          'LINK = {0}'.format(self.compiler.cxx),
     159                          'Makefile.icc')
     160
     161              make('lib', '-f', 'Makefile.icc')
     162              with open('Makefile.lammps', 'w') as fh:
     163                  lapack_blas = (self.spec['lapack'].libs +
     164                                 self.spec['blas'].libs)
     165                  makefile = [
     166                      'user-atc_SYSINC =',
     167                      'user-atc_SYSLIB = {0}'.format(lapack_blas.ld_flags),
     168                      'user-atc_SYSPATH = ']
  >> 169                  fh.write('\n'.join(makefile))

See build log for details:
  /tmp/rpm/spack_stage/spack-stage/spack-stage-hezeAV/lammps-stable_17Nov2016/spack-build.out

$ spack purge; spack clean lammps; spack install lammps %gcc@6 ^openmpi ^openblas
==> Error: lammps does not depend on openblas
```

### PR DESCRIPTION
**Pull request by adamjstewart**
Fixes #3524.

@weijianwen Can you see if this solves your issue?

### CODE REVIEW

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 1]
```diff
@@ -177,15 +177,14 @@ def build_voronoi(self):
                         'compute_voronoi_atom.h')
 
     def build(self, spec, prefix):
-        for pkg in self.supported_packages:
```
Comment (davydden) [inline var/spack/repos/builtin/packages/lammps/package.py]: I think you also change where this is called from, before it was from the root folder and then `with working_dir('lib/atc'):`, now it's from ` with working_dir('src'):` and then `with working_dir('lib/atc'):`.

Comment (davydden) [inline var/spack/repos/builtin/packages/lammps/package.py]: i.e. `voronoi` has `with working_dir('src/VORONOI'):`. So i guess you need to call those functions in place where they were called originally.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lammps/package.py]: That's true. Someone will have to test this out. I'm not sure if `with working_dir` stacks or not, but you're probably right.

**Comment by adamjstewart** [comment]
I think that fixes it. Disclaimer: I don't actually use `lammps` so someone else should test this.

**Comment by adamjstewart** [comment]
Oops

**Comment by weijianwen** [comment]
@adamjstewart Thank you. `spack install lammps %gcc@6 ^openmpi` builds sucessfully. However `spack install lammps+user-atc %gcc@6 ^openmpi ^openblas` fails with the same error message as in #3524 .

**Pull request merged**